apollo-router/src/spec/selection.rs

```diff
@@ -286,70 +244,66 @@ impl Selection {
     }
 }
 
-pub(crate) fn parse_skip_hir(directive: &hir::Directive) -> Option<Skip> {
-    if directive.name() != "skip" {
-        return None;
-    }
-    match directive.argument_by_name("if")? {
-        hir::Value::Boolean(true) => Some(Skip::Yes),
-        hir::Value::Boolean(false) => Some(Skip::No),
-        hir::Value::Variable(variable) => Some(Skip::Variable(variable.name().to_owned())),
-        _ => None,
-    }
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub(crate) struct IncludeSkip {
+    include: Condition,
+    skip: Condition,
 }
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
-pub(crate) enum Skip {
+pub(crate) enum Condition {
     Yes,
     No,
     Variable(String),
 }
 
-impl Skip {
-    pub(crate) fn should_skip(&self, variables: &Object) -> Option<bool> {
-        match self {
-            Skip::Yes => Some(true),
-            Skip::No => Some(false),
-            Skip::Variable(variable_name) => variables
-                .get(variable_name.as_str())
-                .and_then(|v| v.as_bool()),
+impl IncludeSkip {
+    pub(crate) fn parse(directives: &[hir::Directive]) -> Self {
+        let mut include = None;
+        let mut skip = None;
+        for directive in directives {
+            if include.is_none() && directive.name() == "include" {
```

Comment:
Is it specified somewhere that `@skip` or `include` should appear only once? IIRC they should be able to appear multiple times, but have compatible arguments

Reply:
They are specified as:

```graphql
directive @skip(if: Boolean!) on FIELD | FRAGMENT_SPREAD | INLINE_FRAGMENT
directive @include(if: Boolean!) on FIELD | FRAGMENT_SPREAD | INLINE_FRAGMENT
```

Neither is `repeatable`, so having the multiple times at the same location is a validation error

Reply:
alright then